src/wallet/coincontrol.h

```diff
@@ -76,8 +77,24 @@ class CCoinControl
         vOutpoints.assign(setSelected.begin(), setSelected.end());
     }
 
+    void AddKnownCoins(const CInputCoin& coin)
+    {
+        knownCoins.insert(std::make_pair(coin.outpoint, coin));
+    }
+
+    boost::optional<CInputCoin> FindKnownCoin(const COutPoint& outpoint) const
```

Comment:
Another solution is

`bool FindKnownCoin(coinst COutPoint&, CInputCoin& output)`